File: app/src/main/java/br/com/iq/happycalendarandroid/data/TodosContract.java
package br.com.iq.happycalendarandroid.data;

import android.net.Uri;
import android.provider.BaseColumns;

public final class TodosContract {
    //URI section
    public static final String CONTENT_AUTHORITY = "br.com.iq.happycalendarandroid.todosprovider";
    public static final String PATH_TODOS="todos";
    public static final String PATH_CATEGORIES="categories";
    public static final Uri BASE_CONTENT_URI = Uri.parse("content://" + CONTENT_AUTHORITY);

    public String concatContent(String path){
        return "content://" + path;
    }

    public static final class TodosEntry implements BaseColumns {
        public static final Uri CONTENT_URI = Uri.withAppendedPath(BASE_CONTENT_URI, PATH_TODOS);

        // Table name
        public static final String TABLE_NAME = "todos";
        //column (field) names
        public static final String _ID = BaseColumns._ID;
        public static final String COLUMN_TEXT = "text";
        public static final String COLUMN_SPRINT= "sprint";
        public static final String COLUMN_DONE = "done";
        public static final String COLUMN_CATEGORY = "category";
        public static final String COLUMN_BACKLOG = "backlog";
    }

    public static final class CategoriesEntry implements BaseColumns {
        public static final Uri CONTENT_URI = Uri.withAppendedPath(BASE_CONTENT_URI, PATH_CATEGORIES);

        // Table name
        public static final String TABLE_NAME = "categories";
        //column names
        public static final String _ID = BaseColumns._ID;
        public static final String COLUMN_DESCRIPTION = "description";
    }
}

File: app/src/main/java/br/com/iq/happycalendarandroid/data/TodosContract.kt
/*
package br.com.iq.happycalendarandroid.data

import android.net.Uri
import android.provider.BaseColumns

class TodosContract {

    fun concatContent(path: String): String {
        return "content://$path"
    }

    class TodosEntry : BaseColumns {
        companion object {
            val CONTENT_URI = Uri.withAppendedPath(BASE_CONTENT_URI, PATH_TODOS)

            // Table name
            val TABLE_NAME = "todos"
            //column (field) names
            val _ID = BaseColumns._ID
            val COLUMN_TEXT = "text"
            val COLUMN_CREATED = "created"
            val COLUMN_EXPIRED = "expired"
            val COLUMN_DONE = "done"
            val COLUMN_CATEGORY = "category"
        }
    }

    class CategoriesEntry : BaseColumns {
        companion object {
            val CONTENT_URI = Uri.withAppendedPath(BASE_CONTENT_URI, PATH_CATEGORIES)

            // Table name
            val TABLE_NAME = "categories"
            //column names
            val _ID = BaseColumns._ID
            val COLUMN_DESCRIPTION = "description"
        }
    }

    companion object {
        //URI section
        val CONTENT_AUTHORITY = "com.example.todos.todosprovider"
        val PATH_TODOS = "todos"
        val PATH_CATEGORIES = "categories"
        val BASE_CONTENT_URI = Uri.parse("content://$CONTENT_AUTHORITY")
    }
}
*/